lib/jekyll/url.rb

```diff
@@ -84,17 +84,36 @@ def generate_url_from_hash(template)
       end
     end
 
+    # We include underscores in keys to allow for 'i_month' and so forth.
+    # This poses a problem for keys which are followed by and underscore
+    # but the underscore is not part of the key, e.g. '/:month_:day'.
+    # That should be :month and :day, but our key extraction regexp isn't
+    # smart enough to know that so we have to make it an explicit
+    # possibility.
+    def possible_keys(key)
+      if key.end_with?("_")
+        [key, key.chomp("_")]
+      else
+        [key]
+      end
+    end
+
     def generate_url_from_drop(template)
       template.gsub(%r!:([a-z_]+)!) do |match|
-        key = match.sub(":".freeze, "".freeze)
-        unless @placeholders.key?(key)
-          raise NoMethodError, "The URL template key #{key} doesn't exist!"
-        end
-        if @placeholders[key].nil?
-          "".freeze
-        else
-          self.class.escape_path(@placeholders[key])
+        pool = possible_keys(match.sub(":".freeze, "".freeze))
```

Comment:
**casual query:** why do we have to explicitly `freeze` the strings here? :thinking: